pytube/__main__.py

```diff
@@ -185,18 +180,26 @@ def prefetch(self) -> None:
             self.vid_info_url = extract.video_info_url_age_restricted(
                 self.video_id, self.watch_url
             )
+            self.js_url = extract.js_url(self.embed_html)
         else:
             self.vid_info_url = extract.video_info_url(
                 video_id=self.video_id, watch_url=self.watch_url
             )
+            self.js_url = extract.js_url(self.watch_html)
 
         self.initial_data_raw = extract.initial_data(self.watch_html)
         self.initial_data = json.loads(self.initial_data_raw)
 
         self.vid_info_raw = request.get(self.vid_info_url)
-        if not self.age_restricted:
-            self.js_url = extract.js_url(self.watch_html)
+
+        # If the js_url doesn't match the cached url, fetch the new js and update
+        #  the cache; otherwise, load the cache.
+        if pytube.__js_url__ != self.js_url:
```

Comment:
think we could use https://github.com/nficano/pytube/blob/master/pytube/helpers.py#L113 instead?

Reply:
That can't really be used in this case.

The goal of this fix is to prevent repeated network requests to download the `base.js` file from YouTube across a much longer-running application. That helper would only work on a per-object basis, so every new YouTube object will still make the network request.

My implementation caches at the module level -- so every new object will have access to the cached information.

Reply:
so, for example, lets say you have 5 different videos that you want to download:

```python
my_videos = [
   # video links in here
]

for video in my_videos:
    yt = YouTube(video)
    yt.streams.first().download()
```

The line that instantiates the YouTube object will usually *always* fetch the base.js. 
My fix implements a module-level variable that's checked before attempting to download the script.
Using that helper function only caches the function call on a specific object, so it will still download the base js 5 times.

Reply:
Is it safe to say - this is downloading media player from yt with assumption static file wont change in a short duration?

What would happen for long running scripts (?)  e.g. ```for (;;) { download video from $1 }```

Reply:
To some extent, yes. It assumes that the static file won't change in a short duration while also retaining the same url.

It's worth noting that the js url take the form of `/s/player/<random?string>/player_ias.vflset/en_US/base.js`. Currently, whenever I load a page, the random string is 408be03a, which I believe is a sort of hash of the file. 

Any time that the watch_html or embed_html gets loaded, it refers back to the js url to see if the url has changed, and will update the cache if it has.

Reply:
Given the fact that any fundamental changes to this js file will probably break pytube altogether until [extract](https://github.com/nficano/pytube/blob/master/pytube/extract.py#L327) gets updated, I think this is a pretty safe thing to cache.

Reply:
cc: @RONNCC

Reply:
ah sg